src/user/dto/user-response.ts

```diff
@@ -0,0 +1,18 @@
+import { User } from '@prisma/client';
+
+export class UserDto {
+  constructor(
+    public id: number,
+    public email: string,
+    public role: string,
+    public password?: string,
+  ) {}
+
+  public static from(user: User): UserDto {
+    return new UserDto(user.id, user.email, user.role);
+  }
+
+  public static fromWithPassword(user: User): UserDto {
```

Comment:
hmm, I'm wondering if it should't be named `withPasswordFrom` because the line in the code will look like this: `UserDto.withPasswordFrom(user);` - so you would read it as `object of type UserDto with password field created from user object`. It could also be read as if we are getting only password from `user` 😅 But I think it's still better than the current name